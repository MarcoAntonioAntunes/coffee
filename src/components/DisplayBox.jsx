import React from 'react';
import Card from './Card';

const DisplayBox = (props) => {
    const { officeToggle, isDublinPairingsLoading, dublinPairings, isNewYorkPairingsLoading, newYorkPairings } = props;

    return (
        <div className="display-box">
            {officeToggle ? (
                <div className="dublin">
                    <div className="location-title">
                        <h4>Dublin Office</h4>
                    </div>
                    {isDublinPairingsLoading ? (
                        <div className="loader-wrapper">
                            <div className="loader" />
                        </div>
                    ) : (
                            <div>
                                {
                                    dublinPairings.length > 0 ? dublinPairings.map((pair, i) => {
                                        const giver = pair[0];
                                        const receiver = pair[1];
                                        return (
                                            <div key={i} className="card-wrapper">
                                                <Card
                                                    giver={`${giver.name.first} ${giver.name.last}`}
                                                    receiver={`${receiver.name.first} ${receiver.name.last}`}
                                                />
                                            </div>
                                        );
                                    }) : (
                                            <div className="no-users">
                                                <h3>No pairings available</h3>
                                            </div>
                                        )}
                            </div>
                        )}
                </div>
            ) : (
                    <div className="newyork">
                        <div className="location-title">
                            <h4>New York Office</h4>
                        </div>
                        {isNewYorkPairingsLoading ? (
                            <div className="loader-wrapper">
                                <div className="loader" />
                            </div>
                        ) : (
                                <div>
                                    {newYorkPairings.length > 0 ? newYorkPairings.map((pair, i) => {
                                        const giver = pair[0];
                                        const receiver = pair[1];
                                        return (
                                            <div key={i} className="card-wrapper">
                                                <Card
                                                    giver={`${giver.name.first} ${giver.name.last}`}
                                                    receiver={`${receiver.name.first} ${receiver.name.last}`}
                                                />
                                            </div>
                                        );
                                    }) : (
                                            <div className="no-users">
                                                <h3>No pairings available</h3>
                                            </div>
                                        )}
                                </div>
                            )}
                    </div>
                )}
        </div>
    );
};

export default DisplayBox;