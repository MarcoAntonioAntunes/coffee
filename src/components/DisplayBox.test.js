import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import DisplayBox from './DisplayBox';


it('should match the DisplayBox  component to snapshot', () => {
    const component = renderer.create(
        <DisplayBox
            officeToggle={true}
            isDublinPairingsLoading={false}
            dublinPairings={[]}
            isNewYorkPairingsLoading={false}
            newYorkPairings={[]}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render the New york component when the toggle button is off', () => {
    configure({ adapter: new Adapter() });
    const component = shallow(
        <DisplayBox
            officeToggle={false}
            isDublinPairingsLoading={false}
            dublinPairings={[]}
            isNewYorkPairingsLoading={false}
            newYorkPairings={[]}
        />
    );
    const componentChildDiv = component.children().getElements()[0];
    expect(componentChildDiv.props.className).toEqual('newyork');
});


it('should render the Dublin component when the toggle button is off', () => {
    configure({ adapter: new Adapter() });
    const component = shallow(
        <DisplayBox
            officeToggle={true}
            isDublinPairingsLoading={false}
            dublinPairings={[]}
            isNewYorkPairingsLoading={false}
            newYorkPairings={[]}
        />
    );
    const componentChildDiv = component.children().getElements()[0];
    expect(componentChildDiv.props.className).toEqual('dublin');
});