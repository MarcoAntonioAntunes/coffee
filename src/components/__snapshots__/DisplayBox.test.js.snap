// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the DisplayBox  component to snapshot 1`] = `
<div
  className="display-box"
>
  <div
    className="dublin"
  >
    <div
      className="location-title"
    >
      <h4>
        Dublin Office
      </h4>
    </div>
    <div>
      <div
        className="no-users"
      >
        <h3>
          No pairings available
        </h3>
      </div>
    </div>
  </div>
</div>
`;
